tools/settlement/cmd/uphold.go

```diff
@@ -231,13 +241,24 @@ func UpholdUpload(
 			Count:   0,
 		}},
 	}
+
+	maxAmountAny := ctx.Value(appctx.PayoutTxnMaxAmountCTXKey)
+	maxAmount, ok := maxAmountAny.(decimal.Decimal)
+	if !ok {
+		logger.Panic().Err(err).Msg("provided max amount is not a number")
```

Comment:
Nit: Surrounding places to use `Panic`, but the function we are in does return an error. We could return and error and handle it in the caller if we wanted a panic. But also, what value does a panic over error bring here?